specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json

```diff
@@ -1230,10 +1404,87 @@
             "value": "full",
             "name": "Full",
             "description": "Uses the full Lucene query syntax for searches. Search text is interpreted using the Lucene query language which allows field-specific and weighted searches, as well as other advanced features."
+          },
+          {
+            "value": "semantic",
+            "name": "Semantic",
+            "description": "Best suited for queries expressed in natural language as opposed to keywords. Improves precision of search results by re-ranking the top search results using a ranking model trained on the Web corpus."
           }
         ]
       },
-      "description": "Specifies the syntax of the search query. The default is 'simple'. Use 'full' if your query uses the Lucene query syntax."
+      "description": "Specifies the syntax of the search query. The default is 'simple'. Use 'full' if your query uses the Lucene query syntax and 'semantic' if query syntax is not needed."
+    },
+    "Speller": {
+      "type": "string",
+      "enum": [
+        "none",
+        "lexicon"
+      ],
+      "x-ms-enum": {
+        "name": "Speller",
+        "modelAsString": false,
+        "values": [
+          {
+            "value": "none",
+            "name": "None",
+            "description": "Speller not enabled."
+          },
+          {
+            "value": "lexicon",
+            "name": "Lexicon",
+            "description": "Speller corrects individual query terms using a static lexicon for the language specified by the queryLanguage parameter."
+          }
+        ]
+      },
+      "description": "Improve search recall by spell-correcting individual search query terms."
+    },   
+    "Answers": {
+      "type": "string",
+      "enum": [
+        "none",
+        "extractive"
+      ],
+      "x-ms-enum": {
+        "name": "Answers",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "none",
+            "name": "None",
+            "description": "Do not return answers for the query."
+          },
+          {
+            "value": "extractive",
+            "name": "Extractive",
+            "description": "Extracts answer candidates from the contents of the documents returned in response to a query expressed as a question in natural language."
+          }
+        ]
+      },
+      "description": "This parameter is only valid if the query type is 'semantic'. If set, the query returns answers extracted from key passages in the highest ranked documents. The number of answers returned can be configured by appending the |count-<number of answers> option after the answers parameter value. Default is 1."
+    },   
+    "QueryLanguage": {
+      "type": "string",
+      "enum": [
+        "none",
+        "en-us"        
+      ],
+      "x-ms-enum": {
+        "name": "QueryLanguage",
+        "modelAsString": false,
```

Comment:
To confirm, there's no anticipation of future values for this enum?

Reply:
Is the recommendation to use modelAsString for all enums that might have more values in the future?

Reply:
Yes, if you anticipate additions in the future then using `modelAsString:true` is more versioning friendly.